packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -780,6 +815,71 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
     SystemNavigator.pop();
   }
 
+  /// Called at the start of a predictive back gesture.
+  ///
+  /// Returns true if the call should be considered handled.
+  ///
+  /// Currently, this is only used on Android devices that support the
+  /// predictive back feature.
+  Future<bool> _handleStartBackGesture(Map<dynamic, dynamic> arguments) {
+    final PredictiveBackEvent backEvent = PredictiveBackEvent.fromJSON(arguments as Map<String, dynamic>);
+    for (final WidgetsBindingObserver observer in List<WidgetsBindingObserver>.of(_observers)) {
+      if (observer.handleStartBackGesture(backEvent)) {
+        return Future<bool>.value(true);
```

Comment:
So the first observer that returned true gets to handle the event, and observers after that don't even get notified? What happens if I remove that first observer, then all of a sudden another observer starts to receive updates that it didn't receive a begin event for?

Reply:
If possible I think the events probably need an ID like pointer events do.

Reply:
Agreed that it would make this more robust.

Reply:
Instead I did this by directing all events from one gesture to the same observer: https://github.com/flutter/flutter/commit/ac0b088ee463ad197d5dcbc3012fb07f32b680a6